### PR TITLE
SAS Snapshot token creation added

### PR DESCRIPTION
**Pull request by adreed-msft**
This commit adds support for the creation of SAS Snapshot tokens.

I'm unsure about the change of the core file `zc_sas_query_params.go`, as if this is acceptable, it may have to be migrated to other Go azure storage SDKs. However, I also felt it was "dirty" to just use two of the `FormatTimesForSASSigning` function in order to achieve this.

### CODE REVIEW

File: azblob/sas_service.go [hunk 1]
```diff
@@ -30,7 +31,18 @@ type BlobSASSignatureValues struct {
...
-	if v.BlobName == "" {
+	if v.SnapshotTime.IsZero() {
+		resource = "bs"
+
+		perms := &ContainerSASPermissions{}
+		if err := perms.Parse(v.Permissions); err != nil {
+			return SASQueryParameters{}, err
+		}
+		v.Permissions = perms.String()
+		if v.Version == "" {
```
Comment (zezha-msft) [inline azblob/sas_service.go]: isn't this redundant? line 61

File: azblob/zc_sas_query_params.go [hunk 2]
```diff
@@ -1,261 +1,265 @@
-package azblob
```
Comment (zezha-msft) [inline azblob/zc_sas_query_params.go]: what happened here..

File: README.md [hunk 3]
```diff
@@ -33,7 +33,8 @@ This repository contains the open source Blob SDK for Go. The [File SDK][File SD
 |-----------------|---------------------------|----------------------------------------------------------|
 | 2016-05-31      | 0.2.0                     | github.com/Azure/azure-storage-blob-go/2016-05-31/azblob |
 | 2017-07-29      | 0.2.0                     | github.com/Azure/azure-storage-blob-go/2017-07-29/azblob |
-| 2018-03-28      | 0.3.0+                    | github.com/Azure/azure-storage-blob-go/azblob            |
+| 2018-03-28      | 0.3.0 - 0.5.0             | github.com/Azure/azure-storage-blob-go/azblob            |
+| 2018-11-09      | 0.6.0                     | github.com/Azure/azure-storage-blob-go/azblob            |
```
Comment (zezha-msft) [inline README.md]: seems like your branch was based on master. Please submit another PR (by pushing to a branch to this repo and triggering CI), and cherry pick your changes on top of dev for a cleaner view.

**Comment by zezha-msft** [comment]
Could you please add a test for the new SAS?

File: azblob/sas_service.go (hunk 1 above)
Comment (adreed-msft) [reply azblob/sas_service.go]: Actually you're right, it is redundant, I didn't realize while I was restructuring

File: azblob/zc_sas_query_params.go (hunk 2 above)
Comment (adreed-msft) [reply azblob/zc_sas_query_params.go]: Good question, I don't have an answer.

**Comment by zezha-msft** [comment]
duplicated of #124

**Pull request closed without merging**